docs/customizing-makefiles.rst

```diff
@@ -0,0 +1,369 @@
+`Understanding Toolchain Commands <understanding-commands.html>`_ page.
+
+Example 
+-------
+
+Every design in symbiflow has its own Makefile. For example 
+`Counter-test <https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/counter_test/Makefile>`_,  
+`Linux Litex Demo <https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/linux_litex_demo/Makefile>`_, 
+and `Picosoc Demo <https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/picosoc_demo/Makefile>`_ 
+all have there own unique Makefiles for compiling and building respective designs. To understand 
+how to set up a Makefile in symbiflow, lets take a look at a simple Makefile. The following code 
+is based off of the Makefile within `Counter-test <https://github.com/SymbiFlow/symbiflow-examples/blob/master/xc7/counter_test/Makefile>`_ 
+and has been modified slightly for simplicity. Highlighted lines within the code below are of 
+particular interest and will change depending on your specific design elements and hardware. 
+Lines that are not highlighted do not change from design to design and can be copy and pasted 
+into your own Makefile.
+
+.. code-block:: bash
+   :name: makefile-example
+   :emphasize-lines: 3, 4, 5, 6, 9, 10, 22, 25, 28, 31
+   :linenos:
+
+   mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+   current_dir := $(patsubst %/,%,$(dir $(mkfile_path)))
+   TOP:=top
+   VERILOG:=${current_dir}/counter.v 
+   DEVICE  := xc7a50t_test
+   BITSTREAM_DEVICE := artix7
+   BUILDDIR:=build
+
+   PARTNAME:= xc7a35tcpg236-1
+   XDC:=${current_dir}/basys3.xdc 
+   BOARD_BUILDDIR := ${BUILDDIR}/basys3
+
+
+   .DELETE_ON_ERROR:
+
+   all: ${BOARD_BUILDDIR}/${TOP}.bit
+
+   ${BOARD_BUILDDIR}:
+      mkdir -p ${BOARD_BUILDDIR}
+
+   ${BOARD_BUILDDIR}/${TOP}.eblif: | ${BOARD_BUILDDIR}
+      cd ${BOARD_BUILDDIR} && symbiflow_synth -t ${TOP} -v ${VERILOG} -d ${BITSTREAM_DEVICE} -p ${PARTNAME} -x ${XDC} 2>&1 > /dev/null
+
+   ${BOARD_BUILDDIR}/${TOP}.net: ${BOARD_BUILDDIR}/${TOP}.eblif
+      cd ${BOARD_BUILDDIR} && symbiflow_pack -e ${TOP}.eblif -d ${DEVICE} 2>&1 > /dev/null
+
+   ${BOARD_BUILDDIR}/${TOP}.place: ${BOARD_BUILDDIR}/${TOP}.net
+      cd ${BOARD_BUILDDIR} && symbiflow_place -e ${TOP}.eblif -d ${DEVICE} -n ${TOP}.net -P ${PARTNAME} 2>&1 > /dev/null
+
+   ${BOARD_BUILDDIR}/${TOP}.route: ${BOARD_BUILDDIR}/${TOP}.place
+      cd ${BOARD_BUILDDIR} && symbiflow_route -e ${TOP}.eblif -d ${DEVICE} 2>&1 > /dev/null
+
+   ${BOARD_BUILDDIR}/${TOP}.fasm: ${BOARD_BUILDDIR}/${TOP}.route
+      cd ${BOARD_BUILDDIR} && symbiflow_write_fasm -e ${TOP}.eblif -d ${DEVICE}
+
+   ${BOARD_BUILDDIR}/${TOP}.bit: ${BOARD_BUILDDIR}/${TOP}.fasm
+      cd ${BOARD_BUILDDIR} && symbiflow_write_bitstream -d ${BITSTREAM_DEVICE} -f ${TOP}.fasm -p ${PARTNAME} -b ${TOP}.bit
+
+   clean:
+      rm -rf ${BUILDDIR}
+
+
+Adding HDL files to your design 
+--------------------------------
+
+Line 3 in the Makefile shows how to define the name for your top level module. For example if
```

Comment:
Not that I am aware of. I have been looking through several tutorials and it might be possible to link to specific lines of code using a [parsed-literal block](https://stackoverflow.com/questions/44527391/how-to-include-an-internal-reference-in-a-code-block) instead of a code-block. However, that would come with several drawbacks including the loss of line numbers, the inability to highlight lines of interest within the code, and the loss of bash font. I did the next best thing and linked each reference to a line of code to the code block itself. I think this worked out quite well.

Reply:
Ok, I guess that it makes sense to leave this as is than